airflow/providers/google/cloud/transfers/gcs_to_gcs.py

```
@@ -464,7 +466,37 @@ def _copy_single_object(self, hook, source_object, destination_object):
             destination_object,
         )
 
+        self.resolved_source_objects.add(source_object)
+        if not destination_object:
+            self.resolved_target_objects.add(source_object)
+        else:
+            self.resolved_target_objects.add(destination_object)
+
         hook.rewrite(self.source_bucket, source_object, self.destination_bucket, destination_object)
 
         if self.move_object:
             hook.delete(self.source_bucket, source_object)
+
+    def get_openlineage_events_on_complete(self, task_instance):
+        """
+        Implementing _on_complete because execute method does preprocessing on internals.
+        This means we won't have to normalize self.source_object and self.source_objects,
+        destination bucket and so on.
+        """
+        try:
+            from openlineage.client.run import Dataset
+
+            from airflow.providers.openlineage.extractors import OperatorLineage
+        except ImportError:
+            return None
```

I think we should raise this error if the open lineage is enabled. WDYT?

---

Moved catching that to provider.